src/components/micro-frame-sse/parser.ts

```diff
@@ -0,0 +1,86 @@
+interface SSE {
+  lastEventId: string;
+  data: string;
+  type?: string;
+}
+export default (read?: (param: SSE) => [string, string, boolean?]) =>
+  function parser(
+    iterator: AsyncIterator<Buffer | string>
+  ): AsyncIterator<[string, string, boolean?]> {
+    let buf = "";
+    const eventBuf: SSE[] = [];
+
+    function parseEvent(eventStr: string) {
+      let data = "";
+      let id = "";
+      let type;
+      const lines = eventStr.split("\n").filter(Boolean);
+      const idRegEx = /^id: (.*?)$/;
+      const dataRegEx = /^data: (.*?)$/;
+      const typeRegEx = /^type: (.*?)$/;
+      for (const l of lines) {
+        const idRes = idRegEx.exec(l);
+        if (idRes) {
+          id = idRes[1];
+        }
+
+        const typeRes = typeRegEx.exec(l);
+        if (typeRes) {
+          type = typeRes[1];
+        }
+
+        const dataRes = dataRegEx.exec(l);
+        data += dataRes ? dataRes[1] : "";
+      }
+      eventStr = "";
+      return { lastEventId: id, data, type };
+    }
+
+    return {
+      async next() {
+        if (eventBuf.length) {
+          const nextEv = eventBuf.shift() as SSE;
```

Comment:
Probably not a bottleneck here, but generally good for performance to avoid `shift`.